src/HLSMonitorService.ts

```diff
@@ -47,42 +45,115 @@ export class HLSMonitorService {
       exposeRoute: true,
     });
 
-    this.fastify.get("/status", async (request, reply) => {
-      if (!this.hlsMonitor) {
+    this.fastify.get("/monitor/:monitorId/status", async (request, reply) => {
+      if (!this.hlsMonitors.has(request.params.monitorId)) {
         reply.code(500).send({
           status: "error",
           message: "monitor not initialized",
         });
       }
-      const logs = await this.hlsMonitor.getErrors()
+    
+      const logs = await this.hlsMonitors.get(request.params.monitorId).getErrors()
       reply
         .code(200)
         .header('Content-Type', 'application/json; charset=utf-8')
         .send({ logs: logs });
     });
 
-    this.fastify.get("/clear-errors", async (request, reply) => {
-      if (!this.hlsMonitor) {
+    this.fastify.get("/monitor/:monitorId/clear-errors", async (request, reply) => {
+      if (!this.hlsMonitors.has(request.params.monitorId)) {
         reply.code(500).send({
           status: "error",
           message: "monitor not initialized",
         });
       }
-      await this.hlsMonitor.clearErrors();
+      await this.hlsMonitors.get(request.params.monitorId).clearErrors();
       reply
         .code(200)
         .header('Content-Type', 'application/json; charset=utf-8')
         .send({ message: "Cleared errors" });
     });
 
-    this.fastify.get("/streams", async (request, reply) => {
-      if (!this.hlsMonitor) {
+    this,this.fastify.post("/monitor", async (request, reply) => {
+      let uuid = shortUUID.generate();
+      while(this.hlsMonitors.has(uuid)) {
+        uuid = shortUUID.generate();
+      }
+      const body = request.body;
+      let monitor;
+      if(body["stale_limit"]) {
+        monitor = new HLSMonitor(body["streams"], body["stale_limit"]);
+      } else {
+        monitor = new HLSMonitor(body["streams"]);
+      }
+      monitor.create();
+      this.hlsMonitors.set(uuid,monitor)
+      console.log(this.monitors)
+      const rep = {
+        status: "Created a new hls-monitor",
+        streams: body["streams"]
```

Comment:
Just a question but why do you generate the monitor id here and not just in in the `HLSMonitor` constructor when instantiating a new one?

Reply:
I think that is fine considering that it is the HLSMonitorService that manages a set of monitors.